### PR TITLE
MYST-298 Rename nodeKey to providerId

### PR DESCRIPTION
**Pull request by Waldz**

### CODE REVIEW

File: openvpn/management.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ func (management *Management) serveNewConnection(connection net.Conn) {
 func (management *Management) deliverLines() {
 	for {
 		line := <-management.lineReceiver
-		log.Debug(management.logPrefix, "Line delivering: ", line)
+		// log.Debug(management.logPrefix, "Line delivering: ", line)
```
Comment (donce) [inline openvpn/management.go]: Dead code - we should remove it or use `log.Trace` for lower logging level.
Same applies for change at line 138.

File: server/mysterium_api.go [hunk 2]
```diff
@@ -95,8 +96,6 @@ func (mApi *mysteriumAPI) FindProposals(nodeKey string) ([]dto_discovery.Service
 		return nil, err
 	}
 
-	log.Info(mysteriumAPILogPrefix, "FindProposals fetched: ", proposalsResponse.Proposals)
```
Comment (donce) [inline server/mysterium_api.go]: All endpoints adds some logs after it's successful - removing this makes it inconsistent.
We should keep this log or remove all success logs in this class.

File: server/mysterium_api_fake.go [hunk 3]
```diff
@@ -8,44 +8,50 @@ import (
...
+func (client *ClientFake) RegisterIdentity(id identity.Identity, signer identity.Signer) (err error) {
+	client.RegisteredIdentity = id
+	log.Info(mysteriumAPILogPrefix, "Fake newIdentity registered: ", id.Address)
 
 	return nil
 }
 
-func (client *ClientFake) NodeSendStats(nodeKey string, signer identity.Signer) (err error) {
-	log.Info(mysteriumAPILogPrefix, "Node stats sent: ", nodeKey)
+// SendProposalStats heartbeats that service proposal is still active
```
Comment (donce) [inline server/mysterium_api_fake.go]: This comment is false at the moment - we are passing just `providerID` here and no calculations is done in discovery on whether proposal is active or not. All proposals are active at the memotn. What about this?
```
// SendProposalStats heartbeats that provider is still active
```

File: server/mysterium_api_fake.go [hunk 4]
```diff
@@ -55,6 +61,7 @@ func (client *ClientFake) FindProposals(nodeKey string) (proposals []dto_discove
 	return proposals, nil
 }
 
+// SendSessionStats heartbeats that session is still active
```
Comment (donce) [inline server/mysterium_api_fake.go]: This should also include that stats are being sent here about upload and download amounts.

File: openvpn/management.go (hunk 1 above)
Comment (Waldz) [reply openvpn/management.go]: Fixed

File: server/mysterium_api.go (hunk 2 above)
Comment (Waldz) [reply server/mysterium_api.go]: Fixed

File: server/mysterium_api_fake.go (hunk 3 above)
Comment (Waldz) [reply server/mysterium_api_fake.go]: We dont have unique ID for proposals, yet. But intentions is to ping proposal

File: server/mysterium_api_fake.go (hunk 4 above)
Comment (Waldz) [reply server/mysterium_api_fake.go]: Improved

**Comment by Waldz** [comment]
Comments fixed

**Comment by tadovas** [comment]
@donce its on you. LGTM

File: openvpn/management.go [hunk 5]
```diff
@@ -135,7 +135,7 @@ func (management *Management) deliverLines() {
 			lineConsumed = lineConsumed || consumed
 		}
 		if !lineConsumed {
-			log.Warn(management.logPrefix, "Line not delivered: ", line)
+			// log.Warn(management.logPrefix, "Line not delivered: ", line)
```
Comment (shroomist) [inline openvpn/management.go]: why not logging? should this place have a TODO comment?

File: server/mysterium_api.go [hunk 6]
```diff
@@ -64,26 +64,27 @@ func (mApi *mysteriumAPI) RegisterProposal(proposal dto_discovery.ServiceProposa
 	return err
 }
 
-func (mApi *mysteriumAPI) NodeSendStats(nodeKey string, signer identity.Signer) error {
+func (mApi *mysteriumAPI) PingProposal(proposal dto_discovery.ServiceProposal, signer identity.Signer) error {
```
Comment (donce) [inline server/mysterium_api.go]: New method name is false - it states that it acts on "proposal", but implementation acts on "provider".
Naming it `UpdateProviderStatus` seems ok for me.

File: cmd/commands/cli/command_cli.go [hunk 7]
```diff
@@ -130,14 +130,14 @@ func (c *Command) handleActions(line string) {
 
 func (c *Command) connect(argsString string) {
 	if len(argsString) == 0 {
-		info("Press tab to select identity or create a new one. Connect <your-identity> <node-identity>")
+		info("Press tab to select identity or create a new one. Connect <your-id> <provider-id>")
```
[the remark this answers is not in the document]
Comment (Waldz) [reply cmd/commands/cli/command_cli.go]: Fixed

File: server/mysterium_api.go (hunk 6 above)
Comment (Waldz) [reply server/mysterium_api.go]: Tried: `RegisterProvider(proposal)` and `UpdateProviderStatus(proposal)` and `FindProviders(providerId)`, looks weird and reverted.

 I want to make driving change here:
```
 identity -> providerID   -> publish Proposal(providerID, serviceType, ID)
             -> consumerID -> consume Proposal(providerID, serviceType, ID)
```

Comment (donce) [reply server/mysterium_api.go]: Method name should indicate what method does, not what you think it should do in the future.
>Tried: RegisterProvider(proposal) and UpdateProviderStatus(proposal) and FindProviders(providerId), looks weird and reverted.

Why `FindProviders(providerId)` ?? It finds proposals, not providers, so of course it looks weird :D

**Pull request merged**